NEXTBUILTINS.md

```diff
@@ -0,0 +1,261 @@
+# Future built-in decorators
+
+The decorators proposal is intentionally minimal: it only includes the kinds of functionality that are popular with TypeScript "experimental" and Babel "legacy" decorators. There is much more that we can do with these building blocks! This document describes some possible future directions.
+
+**Nothing in this document is part of the MVP proposal; they are ideas for possible future proposals.**
+
+## Use-case-driven built-in decorators for classes
+
+A few common patterns for using decorators could have additional built-in support. Some of these would be possible otherwise, but may be more efficient as built-ins; others require additional support which is not exposed through the three basic built-in decorators.
+
+The decorators in this section are not intended to form any sort of orthogonal, minimal base, but rather to solve practical problems. For a more general approach, see the following section.
+
+### `@tracked`
+
+`@tracked` wraps a public or private field with an accessor, so that, after writing to the field, a particular callback is called. As an example:
+
+```js
+class C {
+  @tracked(f) x;
+}
+```
+
+would behave as:
+
+```js
+class C {
+  #x;
+  get x() { return this.#x; }
+  set x(v) { this.#x = v; f.call(this); }
+}
+```
+
+### `@bound`
+
+`@bound` makes a method auto-bound. By being built-in, it can have the following advantages over the `@bound` decorator described in the main README:
+- Works on private methods, not just public methods.
+- Can be applied to the whole class, rather than just a single method, to apply to all methods in the class.
+- Can be implemented more efficiently, e.g., maintaining stable object shape, avoiding runtime metaprogramming commands, potentially tying into the object representation or the way that calls are done to make allocation lazier.
```

Comment:
I believe (at least some of) the current decorator proposals allow a public field to be created, both when a method is decorated, or on the entire class for all methods, such that no getters/setters are needed.

Reply:
Yes, this is filling a gap left when making the decorators proposal more minimal.